tools/pipeline_perf_test/orchestrator/orchestrator.py

```diff
@@ -42,6 +129,58 @@ def shutdown(self) -> None:
                 print(f"Error stopping/removing Docker container: {e}")
                 print(f"Error output: {e.stderr}")
 
+    def start_monitoring(self, interval: float) -> None:
+        """
+        Monitor Docker container's CPU and memory usage in a background thread using subprocess.
+
+        Args:
+            interval: Time in seconds between polling.
+        """
+        def monitor(container_id: str,
+                    stats: ProcessStats,
+                    stop_event: threading.Event,
+                    interval: float = 1.0):
+            while not stop_event.is_set():
+                try:
+                    cmd = ["docker", "stats", container_id, "--no-stream", "--format",
+                        "{{.Container}} {{.CPUPerc}} {{.MemUsage}}"]
+                    result = subprocess.check_output(cmd, text=True).strip()
+                    if result:
+                        parts = result.split()
+                        # Example: ['11e99006dc11', '87.54%', '21.84MiB', '/', '7.662GiB']
```

Comment:
is the CPU percentage normalized to account for number of cores? In other words, is this expected to between 0-100 or 0-100*#_Of_Cores?

Reply:
docker reports percentage of a single core (so e.g. 800% for a pegged 8 core host would be possible). K8s does  milicores (so 8000m for the same). We normalize both to percentage of a single utilized core between 0-1 (so 8.0 would be 8 pegged cores). I think we might be able to get cpu time from the client lib for docker at least, so maybe we can move it to align better with semantic conventions if that's the case (for now just kind of going with the k8s way of doing it)